src/backends/generic/misc.c

```diff
@@ -156,8 +156,9 @@ const char *Sys_GetBinaryDir(void)
 	SetExecutablePath(exeDir);
 
 	if (exeDir[0] == '\0') {
-		Com_Printf("Couldn't determine executable path. Using ./ instead.\n");
-		Q_strlcpy(exeDir, "./", sizeof(exeDir));
+		getcwd(exeDir, sizeof(exeDir));
+		strcat(exeDir, "/");
+		Com_Printf("Couldn't determine executable path. Using %s instead.\n", exeDir);
```

Comment:
This breaks the build under MacOS and Windows. I think that unistd.h is available under MacOS, so it's properly enough to change the `#if defined` in line 32. For Windows it's more tricky. We can't use the normal `_getcwd()` because it doesn't support wide chars in the returned path, breaking when there are non-ascii characters in it. German umlauts, cyrillic, japanese, chinese, etc. So the way to go is to use `_wgetcwd()` followed by `WideCharToMultiByte(). There is an example in line 67 showing how to use `WideCharToMultiByte()`.